jsonrpc/types.go

```diff
@@ -18,7 +18,7 @@ type transactionOrHash interface {
 
 type transaction struct {
 	Nonce       argUint64      `json:"nonce"`
-	GasPrice    argBig         `json:"gasPrice"`
+	GasPrice    *argBig        `json:"gasPrice"`
```

Comment:
We should add `omitempty` to `GasPrice` as well.

Reply:
I was thinking about this as well. But we are not consistent with other fields either. For example: To, BlockHash, BlockNumber etc.

Reply:
Other fields are common to all transaction types, but `GasPrice` is not set for EIP-1559 transactions, and `GasTipCap` and `GasFeeCap` are not set for legacy transactions, so I would say only those three are optional per say.

Reply:
I would also point out that `gasTipCap` should be `maxPriorityFeePerGas` and `gasFeeCap` should be `maxFeePerGas`. Also `chainId` is missing. Not sure if some of these are being addressed elsewhere